app/Services/PirepService.php

```diff
@@ -132,13 +134,21 @@ public function prefile(User $user, array $attrs): Pirep
             throw new AircraftPermissionDenied($user, $pirep->aircraft);
         }
 
-        // See if this aircraft is at the departure airport
+        // See if this aircraft is valid
         /** @var Aircraft $aircraft */
         $aircraft = $this->aircraftRepo->findWithoutFail($pirep->aircraft_id);
         if ($aircraft === null) {
             throw new AircraftInvalid($aircraft);
         }
 
+        // See if this aircraft is available for flight
+        /** @var Aircraft $aircraft */
+        $aircraft = $this->aircraftRepo->where('id', $pirep->aircraft_id)->where('state', AircraftState::PARKED)->first();
```

Comment:
Is there another setting for this aircraft restriction, I can't remember

Reply:
Like what ?

We are already checking user rights coming from ranks and location of aircraft ... And there are checks for the user itself being at the correct airport.

I do not think that we have any more settings to be checked here. And I think acars is only listing active aircrafts , so this would be the last bit I hope.